packages/engine/src/experiment/package/simple.rs

```diff
@@ -42,62 +63,69 @@ impl SimpleExperiment {
         mut exp_pkg_update_recv: ExpPkgUpdateRecv,
     ) -> Result<()> {
         let max_num_steps = self.config.num_steps;
-        let mut n_sims_steps = HashMap::new();
         let num_sims = self.config.changed_properties.len();
-        for (sim_index, changed_properties) in self.config.changed_properties.iter().enumerate() {
-            let sim_id = sim_index + 1; // We sometimes use 0 as a default/null value, therefore it's not a valid SimulationShortId
-            n_sims_steps.insert(sim_id as SimulationShortId, StepProgress {
-                n_steps: 0,
-                stopped: false,
-            });
-            let msg = ExperimentControl::StartSim {
-                sim_id: sim_id as SimulationShortId,
-                changed_properties: changed_properties.clone(),
-                max_num_steps,
-            };
-            pkg_to_exp.send(msg).await?;
+        let max_sims_in_parallel = self.config.max_sims_in_parallel.unwrap_or(num_sims);
+
+        let mut queued_iter =
+            self.config
+                .changed_properties
+                .iter()
+                .enumerate()
+                .map(|(sim_idx, props)| {
+                    // We sometimes use 0 as a default/null value, therefore it's not a valid
+                    // SimulationShortId
+                    (sim_idx + 1, props)
```

Comment:
`sim_idx + 1` could be cast to a `SimulationShortId` here. It's not an index anyways, since it's off by one.

Reply:
Good suggestion, cheers